src/sys/unix/net.rs

```diff
@@ -1,6 +1,28 @@
 use std::io;
 use std::mem::size_of_val;
-use std::net::SocketAddr;
+use std::net::{self, SocketAddr};
+use std::os::unix::io::FromRawFd;
+
+pub fn tcp_connect(addr: SocketAddr) -> io::Result<net::TcpStream> {
```

Comment:
This could probably still go in a `unix/tcp/...` directory. I think as we are able to just rely on `deref` for *most* of the `std::net` methods, this could still pick up a few specialized functions that Mio has to do.